lib/src/store.dart

```diff
@@ -45,7 +44,7 @@ class Store {
   }
 
   getEntityModelDefinitionFromClass(cls) {
-    return _modelDefinitions[cls]["model"];
+    return _modelDefinitions[cls]["model"]();
```

Comment:
the dynamic stuff is pretty hard to reason about without insider information. What's this returning?

Reply:
This stems from the generated `TestEntity_OBXDefs` map. Right now, I don't see a way to use something class-based here as well, because at that point, we need as much flexibility as possible (type-wise) and it seems like this is not possible with e.g. inheritance, but only maps, where the respective type conversions happen later (that is, e.g. [here](https://github.com/objectbox/objectbox-dart/blob/18-automatic-id-generation/lib/src/store.dart#L52); notice the type conversion involving the template parameter).

Reply:
Implemented in d0cce264fce1babf09e2e3bd158de5e9a42b7788